app/models/description.rb

```diff
@@ -561,11 +580,18 @@ def chg_permission(groups, arg, mode)
   end
 
   def user_made_a_change?(user)
-    self.class.connection.select_value(%(
-      SELECT id FROM #{versioned_table_name}
-      WHERE #{type_tag}_id = #{id} AND user_id = #{user.id}
-      LIMIT 1
-    ))
+    select_manager = arel_select_user_made_a_change(user)
+    self.class.connection.select_value(select_manager.to_sql)
+  end
+
+  # SELECT id FROM #{versioned_table_name}
+  # WHERE #{type_tag}_id = #{id} AND user_id = #{user.id}
+  # LIMIT 1
+  def arel_select_user_made_a_change(user)
+    vtn = Arel::Table.new(versioned_table_name.to_sym)
+    vtn.where(vtn["#{type_tag}_id".to_sym].eq(id).
+              and(vtn[:user_id].eq(user.id))).
+      project(vtn[:id]).take(1)
   end
```

Comment:
Do the version tables have a corresponding AR model?  There is an association, so presumably it must:

  Name#versions

Could this also be rewritten:

  versions.where(user: user).any?

(Note that this method is used only once, and its return value is treated as a simple boolean -- as the question mark suggests it should.)

Reply:
This works, great idea. It's
```ruby
versions.where(user_id: user.id).any?
```